sources/functions/utils

```diff
@@ -108,3 +108,125 @@ function tmp_swap_off() {
     swapoff /.swapfile > /dev/null 2>&1
     rm -f /.swapfile > /dev/null 2>&1
 }
+
+# Will recurse if $pass isn't set or isn't valid
+function password_check() {
+    #If no password set, then get one
+    if [[ -z $pass ]]; then
+        # While password isn't confirmed ...
+        while [[ -z $confirmedPass ]]; do
+            # Get a password
+            if [[ $1 = "whiptail" ]]; then
+                pass=$(whiptail --passwordbox "Enter new password for $user. Leave empty to generate." 9 40 3>&1 1>&2 2>&3)
+                exitstatus=$?
+                if [ "$exitstatus" = 1 ]; then exit 1; fi
+            else
+                echo_query "Enter password for ${user}. (leave empty to generate a password)" "hidden"
+                read -rs 'pass'
+                echo
+            fi
+            # Check the password
+            if [[ $pass = "" ]]; then
+                #If password should be generated, skip the check
+                confirmedPass=true
+            else
+                if [[ $1 = "whiptail" ]]; then
+                    pass2=$(whiptail --passwordbox "Please confirm the password" 9 40 3>&1 1>&2 2>&3)
+                    exitstatus=$?
+                    if [ "$exitstatus" = 1 ]; then exit 1; fi
+                else
+                    echo_query "Please confirm password" "hidden"
+                    read -rs 'pass2'
+                    echo
+                fi
+                if [[ $pass != "$pass2" ]]; then
+                    # Restart if they don't match
+                    if [[ $1 = "whiptail" ]]; then
+                        whiptail --title "Error" --msgbox "Passwords did not match." --ok-button="Try again" 10 50
+                    else
+                        echo_warn "Passwords did not match. Press enter to try again"
+                        read -n 1 -r
+                    fi
+                else
+                    #continue if they're good
+                    confirmedPass=true
+                fi
+            fi
+        done
+        unset confirmedPass
+    fi
+    # If password was empty then it needs to be generated
+    if [[ -z "${pass}" ]]; then
+        echo_progress_start "Generating password"
+        pass="$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c16)"
+        whiptail --title "Generated password" --msgbox "The generated password for $user is:\n$pass \n\nNote this down as this will not be shown again after it is hidden" --ok-button="Hide" 10 70
```

Comment:
I see this change as potentially screwing with some workflows. It's easy to breeze through a bunch of installs atm if you can just go back and grab the generated password later, but I could be wrong on how folks do this

Reply:
I'll add a second whiptail which says "I bet you didn't write that down, did you?" and show it again if need be lol

Reply:
See screen recording in Discord :)